libethereum/TransactionQueue.cpp

```diff
@@ -320,3 +358,43 @@ void TransactionQueue::clear()
 	m_future.clear();
 	m_futureSize = 0;
 }
+
+void TransactionQueue::enqueue(RLP const& _data, h512 const& _nodeId)
+{
+	unique_lock<Mutex> l(x_queue);
+	unsigned itemCount = _data.itemCount();
+	for (unsigned i = 0; i < itemCount; ++i)
+		m_unverified.emplace_back(UnverifiedTransaction(_data[i].data(), _nodeId));
+	m_queueReady.notify_all();
+}
+
+void TransactionQueue::verifierBody()
+{
+	while (!m_aborting)
+	{
+		UnverifiedTransaction work;
+
+		{
+			unique_lock<Mutex> l(x_queue);
+			m_queueReady.wait(l, [&](){ return !m_unverified.empty() || m_aborting; });
```

Comment:
I was suggested by PriorityCompare. Never the less, I'm currently using FIFO to as it does not change much. But the sync mechanism is identical.